someip.lua

```diff
@@ -154,13 +162,20 @@ local function someip_pdu_dissect(buf,pinfo,root)
     if (buf(0,4):uint() == 0xffff8100) and (buf:len() > SOMEIP_HDR_LENGTH)  then
         Dissector.get("sd"):call(buf(SOMEIP_HDR_LENGTH):tvb(),pinfo,root)
     elseif (buf:len() > SOMEIP_HDR_LENGTH) then
-        Dissector.get("data"):call(buf(SOMEIP_HDR_LENGTH):tvb(),pinfo,root)
+        Dissector.get("data"):call(buf(SOMEIP_HDR_LENGTH,-SOMEIP_HDR_LENGTH+8+get_someip_length(buf,pinfo,0)):tvb(),pinfo,root)
     end
 
+    -- Return next SOMEIP packet --
```

Comment:
you should redesign, such that this dissector will call itself as a new protocol layer.